src/main/java/com/epam/service/UserService.java

```diff
@@ -0,0 +1,53 @@
+package com.epam.service;
+
+import com.epam.dao.UserDao;
+import com.epam.model.User;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+public class UserService {
+
+    @Autowired
+    private UserDao userDao;
+
+    public User getUserById(long userId) {
+        return userDao.get(userId);
+    }
+
+    public User getUserByEmail(String email) {
+        return userDao.values().stream()
+                      .filter(user -> user.getEmail().equals(email))
+                      .findFirst()
+                      .orElseThrow(() -> new IllegalStateException("User does not exist with " +
+                              "email: " + email));
+    }
+
+    public List<User> getUsersByName(String name, int pageSize, int pageNum) {
+        int skipCount = (pageNum - 1) * pageSize;
```

Comment:
magic number? btw, why minus 1, not 3?

Reply:
We calculate the number of users to skip for pagination.

pageNum=1, pageSize=5, skipCount=0
pageNum=2, pageSize=5, skipCount=5
pageNum=3, pageSize=5, skipCount=10
pageNum=4, pageSize=5, skipCount=15